### ISSUE TITLE
Possible improvement for reverse geocoding

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#2108]
I notice that a reverse geocoding lookup of locations within boundary http://www.openstreetmap.org/relation/2204948 seems to yield no result, although there are named streets etc. (while outside that relation everything seems to work fine). I assume this may be due to the fact that no "place= (suburb, etc.)" tag is assigned (?).

If this is true, we may want to allow for similar mapping cases like this in our reverse geocoding code?

**Comment by sonora** [comment]
PS: It may or may not be related, but after excessive testing in a row I get enclosed exception.
[exception.log..pdf](https://github.com/osmandapp/Osmand/files/76431/exception.log.pdf)

**Comment by sonora** [comment]
PS 2: A route from the area in question finds the street names all right (in the route description).

Also: Extensive search shows there seem to be a few other areas behaving like this, it may not have to do with the presence of a place tag.

**Comment by vshcherb** [comment]
I think it is solved already, I don't see any issue.
https://www.dropbox.com/s/flt99zngtytpwge/Screenshot%202016-01-17%2023.45.37.png?dl=0
https://www.dropbox.com/s/iu87z155req08sc/Screenshot%202016-01-17%2023.45.14.png?dl=0

**Comment by sonora** [comment]
Yes, looks like it works now. Took a map update to work, that's why I did not find out myself - Thx!

**Comment by sonora** [comment]
This one drives me crazy ... after having worked immediately after the (full Spain) map upgrade, the issue seems to have come back. Now I found out this:

I also have an old Africa map of June 5, 2015, which covers the area, and it works flawlessly. Then I tried the "Canary Islands" map, and it also has no issues. So I now think that somehow we are approaching an index size limit with the full Spain map (it is around 1 GB), which causes the reverse lookup to break or rather become extremely slow under some circumstances.

PS: I may be wrong, but I have a feeling that when the search hangs in "Looking up address" in this fashion (with the full Spain map), the phone gets a little hot, so the CPU maybe be very busy, like hung in a loop. (Maybe the loop does not hit the breakpoint or something and the search for some streets continues in all of Spain or similar ...)

PS2:  Interesting also; When it works, the results from the full Spain map seem to contain the village  (in some cases suburb  in parentheses), while the results from the Canary Islands map are always MUCH faster, but seem to only state the street name.
